internal/pkg/deploy/cloudformation/app.go

```diff
@@ -56,6 +56,12 @@ func (cf CloudFormation) DeployApp(in *deploy.CreateAppInput) error {
 
 func (cf CloudFormation) UpgradeApplication(in *deploy.CreateAppInput) error {
 	appConfig := stack.NewAppStackConfig(in)
+	appStack, err := cf.cfnClient.Describe(appConfig.StackName())
+	if err != nil {
+		return fmt.Errorf("get existing application infrastructure stack: %w", err)
+	}
+	in.DNSDelegationAccounts = stack.DNSDelegatedAccountsForStack(appStack.SDK())
+	appConfig = stack.NewAppStackConfig(in)
```

Comment:
OK cool just to confirm, we don't need to update `DelegateDNSPermissions` right? I looked at the code and these changes seem good to me but wanted to validate.

Reply:
No i don't think so...since in `DelegateDNSPermissions` we add the new account to createAppInput.AppDNSDelegatedAccounts